File: AccountTransferMvpApp/AccountTransferMvpApp/Utils/ServiceLocator.m
#import "ServiceLocator.h"
#import "PresenterTypeEnum.h"
#import "LocalAccountRepository.h"
#import "ILocalAccountRepository.h"
#import "RemoteAccountRepository.h"
#import "AccountTransferPresenter.h"


@implementation ServiceLocator {

}

+ (id)resolve:(PresenterTypeEnum)presenterTypeEnum {
    switch (presenterTypeEnum) {
        case AppLaunch:
            return nil;
        case AccountTransfer:
            return [self createAccountTransferPresenter];
    }
 return nil;
}

+ (id)createAccountTransferPresenter {
    id<IRemoteAccountRepository> remoteAccountRepository = [[RemoteAccountRepository alloc] init];
    id<ILocalAccountRepository> localAccountRepository = [[LocalAccountRepository alloc] init];
    return [[AccountTransferPresenter alloc] initWithRemote:remoteAccountRepository andLocal:localAccountRepository];
}
@end

File: AccountTransferMvpApp/src/com/gaddzeit/utils/ServiceLocator.java
package com.gaddzeit.utils;

import com.gaddzeit.presenters.AccountTransferPresenter;
import com.gaddzeit.presenters.IPresenter;
import com.gaddzeit.repositories.ILocalAccountRepository;
import com.gaddzeit.repositories.IRemoteAccountRepository;
import com.gaddzeit.repositories.LocalAccountRepository;
import com.gaddzeit.repositories.RemoteAccountRepository;

public class ServiceLocator {
    public static IPresenter resolve(PresenterTypeEnum presenterTypeEnum) {
        switch(presenterTypeEnum) {
            case AppLaunch:
                return null;
            case AccountTransfer:
                return createAccountTransferPresenter();
        }
        return null;
    }

    private static IPresenter createAccountTransferPresenter() {
        IRemoteAccountRepository remoteAccountRepository = new RemoteAccountRepository();
        ILocalAccountRepository localAccountRepository = new LocalAccountRepository();
        return new AccountTransferPresenter(remoteAccountRepository, localAccountRepository);
    }
}
